ring/ring_test.go

```diff
@@ -208,26 +217,16 @@ func TestDoBatchWithOptionsContextCancellation(t *testing.T) {
 
 	callback := func(InstanceDesc, []int) error { return nil }
 	desc := &Desc{Ingesters: generateRingInstances(NewRandomTokenGeneratorWithSeed(0), numInstances, numZones, numTokens)}
-	r := Ring{
-		cfg: Config{
-			HeartbeatTimeout:     time.Hour,
-			ZoneAwarenessEnabled: true,
-			SubringCacheDisabled: true,
-			ReplicationFactor:    numZones,
-		},
-		ringDesc:              desc,
-		ringTokens:            desc.GetTokens(),
-		ringTokensByZone:      desc.getTokensByZone(),
-		ringInstanceByToken:   desc.getTokensInfo(),
-		ringZones:             getZones(desc.getTokensByZone()),
-		instancesCountPerZone: desc.instancesCountPerZone(),
-		shuffledSubringCache:  map[subringCacheKey]*Ring{},
-		strategy:              NewDefaultReplicationStrategy(),
-		lastTopologyChange:    time.Now(),
-	}
+	r := newRingForTesting(Config{
+		HeartbeatTimeout:     time.Hour,
+		ZoneAwarenessEnabled: true,
+		SubringCacheDisabled: true,
+		ReplicationFactor:    numZones,
+	}, true)
+	r.setRingStateFromDesc(desc, false, false, true)
```

Comment:
Why true for `updateReadOnlyInstances`?

Reply:
Also here the code was not doing that before. Let me recheck all places again. I suspect I changed my mind about it mid-work (don't remember), so it's not consistent now.

Reply:
@pstibrany yes, there were some more call sites, but I wanted to hear from you first :D